src/layer/pooling.cpp

```diff
@@ -99,6 +102,83 @@ int Pooling::forward(const Mat& bottom_blob, Mat& top_blob, const Option& opt) c
         return 0;
     }
 
+    if (adaptive_pooling)
+    {
+        top_blob.create(out_w, out_h, channels, elemsize, opt.blob_allocator);
+        if (top_blob.empty())
+            return -100;
```

Comment:
Is it possible that explicit padding (pad_left pad_right ...) could be applied before pooling section ?

Reply:
According PyTorch implementation, it do not need padding params. [AdaptiveAvgPoolKernel](https://github.com/pytorch/pytorch/blob/690eaf9c43f3f71051d90e9f47f25b014ce84681/aten/src/ATen/native/cpu/AdaptiveAvgPoolKernel.cpp#L33)